deltachat-ios/Intents/OpenWebXDCAppIntent.swift

```diff
@@ -0,0 +1,130 @@
+    }
+
+    @MainActor
+    init(accountId: Int, chat: DcChat, msg: DcMsg) {
+        self.id = .init(accountId: accountId, chatId: msg.chatId, msgId: msg.id)
+        let dict = msg.getWebxdcInfoDict()
+        let document = dict["document"] as? String ?? ""
+        let name = dict["name"] as? String ?? "ErrName" // name should not be empty
+        self.title = document.isEmpty ? name : "\(document) – \(name)"
+        self.subtitle = chat.name.isEmpty ? nil : "shared with " + chat.name
+        self.icon = (dict["icon"] as? String).flatMap(msg.getWebxdcBlob(filename:))
+    }
+
+    var title: String
+    var subtitle: String?
+    var icon: Data?
+
+    static var typeDisplayRepresentation = TypeDisplayRepresentation(name: "Shared WebXDC App")
+
+    var displayRepresentation: DisplayRepresentation {
+        DisplayRepresentation(
+            title: LocalizedStringResource(stringLiteral: title),
+            subtitle: subtitle.map { .init(stringLiteral: $0) },
+            image: icon.map { .init(data: $0) }
+        )
+    }
+
+    @MainActor
+    static var defaultQuery = DefaultQuery()
+    struct DefaultQuery: EntityQuery {
+        @MainActor
+        func entities(for identifiers: [WebXDCAppEntity.ID]) async throws -> [WebXDCAppEntity] {
+            Dictionary(grouping: identifiers, by: \.accountId).flatMap { accountId, identifiersForAccount in
+                let dcContext = DcAccounts.shared.get(id: accountId)
+                return identifiersForAccount.map {
+                    WebXDCAppEntity(accountId: accountId, chat: dcContext.getChat(chatId: $0.chatId), msg: dcContext.getMessage(id: $0.msgId))
+                }
+            }
+        }
+
+
+        @MainActor
+        @available(iOS 17, *)
+        var onlySuggestWebxdcApp: WebXDCAppEntity? {
+            get { _onlySuggestWebxdcApp }
+            set { _onlySuggestWebxdcApp = newValue }
+        }
+        /// Mutating this in iOS 16 crashes the app :)
+        @MainActor private var _onlySuggestWebxdcApp: WebXDCAppEntity? {
+            didSet { OpenWebXDCAppShortcuts.updateAppShortcutParameters() }
+        }
+
+        @MainActor
+        func suggestedEntities() async throws -> [WebXDCAppEntity] {
+            if #available(iOS 17, *), let onlySuggestWebxdcApp {
+                return [onlySuggestWebxdcApp]
+            } else {
+                return DcAccounts.shared.getAll().flatMap { accountId in
+                    let dcContext = DcAccounts.shared.get(id: accountId)
+                    let webxdcMsgIds = dcContext.getChatMedia(chatId: 0, messageType: DC_MSG_WEBXDC, messageType2: 0, messageType3: 0)
+                    return webxdcMsgIds.map { webxdcMsgId in
+                        let msg = dcContext.getMessage(id: webxdcMsgId)
+                        let chat = dcContext.getChat(chatId: msg.chatId)
+                        return WebXDCAppEntity(accountId: accountId, chat: chat, msg: msg)
+                    }
+                }
+            }
```

Comment:
If you have multiple accounts with many xdcs it is not easy to find the one you are looking for.
Atleast sorted by recency would only partially work with how I imagine .flatMap to work. Like for first account it is fine, but for the second account you would need to scroll past all entries for the first account, wouldn't you? That is if there is no limit on the amount of suggestions shown?

I would suggest either making custom chooser ui if thats possible (select account, then webxdc from media list or even chatlist), or giving power users a way to find out account id and message id to construct the deep link themselves (or give them a button to "copy deep link for shortcut" in the 3dot menu of the webxdc or in the message info of the webxdc's message).

Reply:
This is the UI you get in Shortcuts app:

![image](https://github.com/user-attachments/assets/575b69e6-bb9d-46af-836f-34f4f7bb2f67)